src/Microsoft.AspNet.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -174,10 +113,12 @@ public OAuthHandler(HttpClient backchannel)
             requestMessage.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
             requestMessage.Content = requestContent;
             var response = await Backchannel.SendAsync(requestMessage, Context.RequestAborted);
-            response.EnsureSuccessStatusCode();
-            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
-
-            return new OAuthTokenResponse(payload);
+            if (response.IsSuccessStatusCode)
```

Comment:
Yes, removing the catch-alls was expected to uncover suppressed exceptions and destabilize things a bit. Dealing with those kinds of issues can wait until after this PR. We'll want to look a them on a case-by-case basis. In the meantime if there are test issues we can update them or disable them.

The change above re-suppresses an error that we were trying to uncover and makes it harder to notice and handle properly in the future.